### PR TITLE
[flang-drv][NFC] Introduce the flang namespace

### PR DESCRIPTION
**Pull request by banach-space**
This patch introduces the flang namespace in the Flang driver. The new
namespace is used to guard everything that belongs to the Flang driver.
This way we are guaranteed to avoid name clashes with Clang.

I've also renamed ExecuteCompilerInvocation.h as Utils.h. This is for:
* consistency with Clang
* ExecuteCompilerInvocation.h/Utils.h contains only basic utility
methods, so Utils.h feels more fitting.

### CODE REVIEW

File: flang/lib/FrontendTool/ExecuteCompilerInvocation.cpp [hunk 1]
```diff
@@ -14,8 +14,13 @@
 #include "flang/Frontend/CompilerInstance.h"
 #include "flang/Frontend/FrontendAction.h"
 #include "flang/Frontend/FrontendActions.h"
+#include "flang/FrontendTool/Utils.h"
```
Comment (CarolineConcatto) [inline flang/lib/FrontendTool/ExecuteCompilerInvocation.cpp:17]: Shouldn't this be the first library?

File: flang/include/flang/FrontendTool/Utils.h [hunk 2]
```diff
@@ -12,24 +11,28 @@
...
+
 class CompilerInstance;
 class FrontendAction;
 
 /// Construct the FrontendAction of a compiler invocation based on the
 /// options specified for the compiler invocation.
 ///
 /// \return - The created FrontendAction object
-std::unique_ptr<FrontendAction> CreateFrontendAction(CompilerInstance &CI);
+// std::unique_ptr<FrontendAction> CreateFrontendAction(CompilerInstance &CI);
```
Comment (CarolineConcatto) [inline flang/include/flang/FrontendTool/Utils.h:28]: why is this line commented?

Comment (banach-space) [reply flang/include/flang/FrontendTool/Utils.h:28]: It shouldn't, sorry!

File: flang/lib/FrontendTool/ExecuteCompilerInvocation.cpp (hunk 1 above)
Comment (banach-space) [reply flang/lib/FrontendTool/ExecuteCompilerInvocation.cpp:17]: Tricky question :) It should be if it was the main module header: https://llvm.org/docs/CodingStandards.html#include-style. But then the name suggests that it isn't!

I will move it to the top, though for me this is an evidence that `FlangFrontendTool` is just a bit awkward and perhaps not needed?

**Comment by banach-space** [comment]
I've rebased this PR on top of `flang-driver-dev`.

File: flang/lib/FrontendTool/ExecuteCompilerInvocation.cpp (hunk 1 above)
Comment (CarolineConcatto) [reply flang/lib/FrontendTool/ExecuteCompilerInvocation.cpp]: Well, if you changed the name of the file to be consistent with clang, so it is to have Frontendtool, no?
For me it is always tricky when should we stop our inspiration. But either way is fine. You can leave this way or the other.

**Pull request merged**